tekton/ci/shared/go_coverage.yaml

```diff
@@ -0,0 +1,215 @@
+      description: coverage percentage below which to mark the coverage test as failed.
+      default: 0
+    - name: uploadGcsBucket
+      description: GCS bucket to upload artifacts to.
+  workspaces:
+    - name: source
+      description: Location where the git repo will be installed.
+    - name: github-token
+      description: |
+        A secret workspace where the github personal access token resides.
+        The token should be storage as a stringData with key "token" and
+        value "your access token".
+    - name: credentials
+  tasks:
+    - name: check-name-matches
+      taskRef:
+        name: check-name-matches
+      params:
+        - name: gitHubCommand
+          value: $(params.gitHubCommand)
+        - name: checkName
+          value: $(params.jobName)
+    - name: split-full-repo-name
+      when:  # implicit dependency on the check tasks
+      - input: $(tasks.check-name-matches.results.check)
+        operator: in
+        values: ["passed"]
+      params:
+        - name: package
+          value: $(params.package)
+      taskSpec:
+        params:
+          - name: package
+            description: The GitHub org/repo
+        results:
+          - name: repoOwner
+            description: Owner of the repo.
+          - name: repoName
+            description: Name of the repo.
+        steps:
+          - name: split-name
+            image: ubuntu
+            script: |
+              printf '$(params.package)' | awk -F/ '{printf($1)}' | tee /tekton/results/repoOwner
+              printf '$(params.package)' | awk -F/ '{printf($2)}' | tee /tekton/results/repoName
+    - name: run-go-coverage
+      runAfter: ['split-full-repo-name']
+      taskRef:
+        name: go-coverage
+      params:
+        - name: pullRequestNumber
+          value: $(params.pullRequestNumber)
+        - name: gitRevision
+          value: $(params.gitRevision)
+        - name: buildUUID
+          value: $(params.buildUUID)
+        - name: repoName
+          value: $(tasks.split-full-repo-name.results.repoName)
+        - name: repoOwner
+          value: $(tasks.split-full-repo-name.results.repoOwner)
+        - name: jobType
+          value: $(params.jobType)
+        - name: jobName
+          value: $(params.jobName)
+        - name: postSubmitGcsBucket
+          value: $(params.postSubmitGcsBucket)
+        - name: postSubmitJobName
+          value: $(params.postSubmitJobName)
+        - name: covTarget
+          value: $(params.covTarget)
+        - name: profileName
+          value: $(params.profileName)
+        - name: covThresholdPercentage
+          value: $(params.covThresholdPercentage)
+      workspaces:
+        - name: source
+          workspace: source
+        - name: github-token
+          workspace: github-token
+    - name: upload-artifacts
+      runAfter: ['run-go-coverage']
+      taskRef:
+        name: gcs-upload
+        bundle: gcr.io/tekton-releases/catalog/upstream/gcs-upload:0.1
```

Comment:
There is version 0.3 of this in the catalog - any reason for using 0.1?
Version 0.3 default behaviour is safer https://github.com/tektoncd/catalog/tree/main/task/gcs-upload/0.3#parameters

Reply:
I don't think so. Just an oversight. I will move it to 0.3

Reply:
done